src/main/java/io/vertx/ext/stomp/AuthenticationHandler.java

```diff
@@ -0,0 +1,28 @@
+package io.vertx.ext.stomp;
+
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+
+/**
+ * Handler called when an authentication is required by the server. Authentication happen during the {@code CONNECT /
+ * CONNECTED} phase. The {@code CONNECT} frame should includes the {@code login} and {@code passcode} header
+ * (configured from the client option. The login and passcode are passed to the handler than can check whether or not
+ * the access is granted. When the decision has been made, it must called the {@code resultHandler} with the value
+ * {@code true} if the access if granted, {@code false} otherwise.
+ */
+@VertxGen
+public interface AuthenticationHandler {
```

Comment:
Would be better to plugin to the existing AuthProvider stuff so that users can use different pre-existing auth providers with their STOMP server too, instead of having to manage auth in a different way.